### PR TITLE
Add unit tests for posts and comments

### PR DESCRIPTION
**Pull request by stavby**

### CODE REVIEW

File: utils/types.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export type TypeWithId<T> = T & { _id?: string };
```
Comment (yarondlugin) [inline utils/types.ts]: why is _id not required?

File: controllers/comments.ts [hunk 2]
```diff
@@ -3,23 +3,40 @@ import httpStatus from 'http-status';
 import { isValidObjectId } from 'mongoose';
 import { commentModel } from '../models/comments';
 import { formatValidationError } from '../utils/formatValidationError';
+import { postModel } from '../models/posts';
 
 export const createComment = async (request: Request, response: Response, next: NextFunction) => {
 	const data = request.body;
 
 	try {
+		const { postId } = data;
```
Comment (yarondlugin) [inline controllers/comments.ts]: move this and the post validation condition out of the try section

File: controllers/comments.ts [hunk 3]
```diff
@@ -3,23 +3,40 @@ import httpStatus from 'http-status';
...
+
+		if (!postId || !isValidObjectId(postId)) {
+			response.status(httpStatus.BAD_REQUEST).send(`Invalid post id "${postId || '(empty)'}"`);
+			return;
+		}
+		const postExists = await postModel.exists({ _id: postId });
+		if (!postExists) {
+			response.status(httpStatus.BAD_REQUEST).send(`Post with id ${postId} doesn't exist`);
```
Comment (yarondlugin) [inline controllers/comments.ts]: change to NOT_FOUND status

File: tests/comments.test.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+import { app } from '../index';
+import { closeDB } from '../services/db';
+import { Comment } from '../models/comments';
+
+afterAll(() => {
+	closeDB();
+});
+
+describe('Comments API', () => {
+	let commentId: string;
```
Comment (yarondlugin) [inline tests/comments.test.ts:13]: Isn't this bad practice? or it's ok / unavoidable when writing tests

File: tests/comments.test.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+					sender: 'testuser',
+					content: 'This is a test comment',
+					postId: 'invalid-post-id',
+				})
+				.expect(httpStatus.BAD_REQUEST);
+
+			expect(response.text).toBe('Invalid post id "invalid-post-id"');
+		});
+
+		it('should return 400 for getting comments with invalid post id filter', async () => {
```
Comment (yarondlugin) [inline tests/comments.test.ts:132]: maybe add one for non existing post id? and expect to get an empty result

File: utils/types.ts (hunk 1 above)
Comment (stavby) [reply utils/types.ts]: because when first creating the object it doesn't have an id

File: controllers/comments.ts (hunk 3 above)
Comment (stavby) [reply controllers/comments.ts]: I don't think it's right, 404 means the resource you've requested wasn't found, but this is a part of a resource you're creating

File: tests/comments.test.ts (hunk 4 above)
Comment (stavby) [reply tests/comments.test.ts:13]: I'm not sure what you mean, this is a variable to be set once we've created a comment through the test suite, and then used for other tests like GET and UPDATE

File: utils/types.ts (hunk 1 above)
Comment (yarondlugin) [reply utils/types.ts]: when does this happen? do you mean when parsing the user input into an object, before sending it to mongoose?

Comment (yarondlugin) [reply utils/types.ts]: in that case maybe they should be seperate types

File: controllers/comments.ts (hunk 3 above)
Comment (yarondlugin) [reply controllers/comments.ts]: ur right, i was confused

File: utils/types.ts (hunk 1 above)
Comment (stavby) [reply utils/types.ts]: ok sure

**Review by yarondlugin — APPROVED** [review]
good good

**Pull request merged**